shared/src/main/scala/amf/core/AMFSerializer.scala

```diff
@@ -114,31 +112,36 @@ class AMFSerializer(unit: BaseUnit,
     w.toString
   }
 
-  protected def findDomainPlugin(): Option[AMFDocumentPlugin] =
-    AMFPluginsRegistry.documentPluginForVendor(vendor).find { plugin =>
-      plugin.documentSyntaxes.contains(mediaType) && plugin.canUnparse(unit)
-    } match {
-      case Some(domainPlugin) =>
-        Some(domainPlugin)
-      case None => AMFPluginsRegistry.documentPluginForMediaType(mediaType).find(_.canUnparse(unit))
-    }
 
-  private def getDomainPlugin: AMFDocumentPlugin =
-    findDomainPlugin().getOrElse {
+  private def getRenderPlugin: AMFRenderPlugin = {
+    val renderPlugin = env.registry.plugins.renderPlugins.sorted.find(_.applies(RenderInfo(unit, vendor, mediaType)))
```

Comment:
The fact that we have to call `sorted` all the time by hand means that the `renderPlugins` list should be a SortedList

Reply:
This is a valid point, and also applies to parsing plugins for example.